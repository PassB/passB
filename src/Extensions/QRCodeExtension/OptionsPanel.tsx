import {FormControl, FormControlLabel, FormLabel, Grid, Input, List, ListItem, Radio, RadioGroup} from 'material-ui';
import {withStyles, ClassProps} from 'material-ui/styles';
import * as React from 'react';
import {QRCode} from 'react-qr-svg';
import {OptionPanelProps} from 'Options/OptionsReceiver';
import {Level, Options} from './QRCodeExtension';

const styles = {
  qrcode: {
    height: '150px',
  },
};

export const OptionsPanel = withStyles<OptionPanelProps<Options>>(styles)(
  ({options, updateOptions, classes}: OptionPanelProps<Options> & ClassProps<typeof styles>): JSX.Element => (
    <Grid container={true} direction="row" justify="space-between" align="center" spacing={0}>
      <Grid item={true}>
        <List>
          <ListItem>
            <FormControl>
              <FormLabel>Error correction level:</FormLabel>
              <RadioGroup
                value={options.level}
                row={true}
                onChange={(event: React.InputHTMLAttributes<HTMLInputElement>, value: Level) => updateOptions({
                  ...options,
                  level: value,
                })}
              >

                <FormControlLabel value="L" control={<Radio/>} label="L"/>
                <FormControlLabel value="M" control={<Radio/>} label="M"/>
                <FormControlLabel value="Q" control={<Radio/>} label="Q"/>
                <FormControlLabel value="H" control={<Radio/>} label="H"/>

              </RadioGroup>
            </FormControl>
          </ListItem>
          <ListItem>
            <FormControl>
              <FormLabel>Background color:</FormLabel>
              <Input
                inputProps={{
                  onChange: (event: React.ChangeEvent<HTMLInputElement>) => updateOptions({
                    ...options,
                    bgColor: event.target.value,
                  }),
                }}
                type="color"
                value={options.bgColor}
              />
            </FormControl>
          </ListItem>
          <ListItem>
            <FormControl>
              <FormLabel>Foreground color:</FormLabel>
              <Input
                inputProps={{
                  onChange: (event: React.ChangeEvent<HTMLInputElement>) => updateOptions({
                    ...options,
                    fgColor: event.target.value,
                  }),
                }}
                type="color"
                value={options.fgColor}
              />
            </FormControl>
          </ListItem>
        </List>
      </Grid>
      <QRCode
        className={classes.qrcode}
        {...options}
        value="example. happy testing ðŸ˜€"
      />
    </Grid>
  ));
