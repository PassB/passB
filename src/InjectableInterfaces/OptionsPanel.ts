import {ComponentType} from 'react';
import {MapTypeAllowedData, TypedMap} from 'State/Types/TypedMap';

export interface OptionPanelProps<OptionType extends MapTypeAllowedData<OptionType>> {
  options: TypedMap<OptionType>;
  updateOptions: (newOptions: TypedMap<OptionType>) => void;
}

export type OptionsPanelType<OptionType extends MapTypeAllowedData<OptionType>> = ComponentType<OptionPanelProps<OptionType>>;
