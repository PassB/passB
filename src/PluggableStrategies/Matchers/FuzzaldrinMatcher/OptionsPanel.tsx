import {TextField} from 'material-ui';
import * as React from 'react';
import {OptionPanelProps} from 'Options/OptionsReceiver';
import {Options} from './FuzzaldrinMatcher';

export function OptionsPanel({options, updateOptions}: OptionPanelProps<Options>): JSX.Element {
  return (
    <div>
      <TextField
        label="Minimum score:"
        helperText=""
        value={options.minScore}
        onChange={(e: any) => updateOptions({ // tslint:disable-line:no-any
          // required until material-ui get their typings right
          // this will be finally fixed with beta v1.0.0-beta.17
          ...options,
          minScore: Math.max(0, (e as React.ChangeEvent<HTMLInputElement>).target.valueAsNumber),
        })}
        type="number"
      />
      <br/>
      <TextField
        label="Maximum visible results:"
        helperText="to show all results, enter '0'"
        value={options.maxResults}
        onChange={(e: any) => updateOptions({ // tslint:disable-line:no-any
          // required until material-ui get their typings right
          ...options,
          maxResults: Math.max(0, (e as React.ChangeEvent<HTMLInputElement>).target.valueAsNumber),
        })}
        type="number"
      />
    </div>
  );
}
