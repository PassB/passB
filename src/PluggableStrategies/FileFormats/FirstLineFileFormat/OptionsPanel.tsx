import {FormControl, FormControlLabel, FormLabel} from 'material-ui';
import {default as Radio, RadioGroup} from 'material-ui/Radio';
import * as React from 'react';
import {OptionPanelProps} from 'InjectableInterfaces/OptionsPanel';
import {Options, UsernameStyle} from './FirstLineFileFormat';

export function OptionsPanel({options, updateOptions}: OptionPanelProps<Options>): JSX.Element {
  return (
    <div>
      <FormControl>
        <FormLabel>Get username from:</FormLabel>
        <RadioGroup
          name="FirstLineFileFormat_usernameStyle"
          value={options.get('usernameStyle')}
          onChange={(event: React.InputHTMLAttributes<HTMLInputElement>, value: UsernameStyle) =>
            updateOptions(options.set('usernameStyle', value))
          }
        >
          <FormControlLabel value="None" control={<Radio/>} label="None"/>
          <FormControlLabel value="SecondLine" control={<Radio/>} label="SecondLine"/>
          <FormControlLabel value="LastPathPart" control={<Radio/>} label="LastPathPart"/>
        </RadioGroup>
      </FormControl>
    </div>
  );
}
