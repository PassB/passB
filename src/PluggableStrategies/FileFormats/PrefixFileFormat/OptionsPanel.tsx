import {Checkbox, FormControl, FormControlLabel, TextField} from 'material-ui';
import * as React from 'react';
import {OptionPanelProps} from 'InjectableInterfaces/OptionsPanel';
import {Options} from './PrefixFileFormat';

export function OptionsPanel({options, updateOptions}: OptionPanelProps<Options>): JSX.Element {
  return (
    <FormControl fullWidth={true}>
      <FormControlLabel
        id="passwordFirstLine"
        control={
          <Checkbox
            checked={options.get('passwordFirstLine')}
            onChange={
              (event: React.ChangeEvent<HTMLInputElement>, isChecked: boolean) => updateOptions(options.set('passwordFirstLine', isChecked))
            }
          />
        }
        label="Password is in first line (without prefix)"
      />
      {!options.get('passwordFirstLine') &&
      <TextField
        id="passwordPrefix"
        label="password prefix"
        value={options.get('passwordPrefix')}
        onChange={
          (event: React.ChangeEvent<HTMLInputElement>) => updateOptions(options.set('passwordPrefix', event.target.value))
        }
      />
      }
      <TextField
        id="usernamePrefix"
        label="username prefix"
        value={options.get('usernamePrefix')}
        onChange={
          (event: React.ChangeEvent<HTMLInputElement>) => updateOptions(options.set('usernamePrefix', event.target.value))
        }
      />
      <FormControlLabel
        id="trimWhitespace"
        control={
          <Checkbox
            checked={options.get('trimWhitespace')}
            onChange={
              (event: React.ChangeEvent<HTMLInputElement>, isChecked: boolean) =>  updateOptions(options.set('trimWhitespace', isChecked))
            }
          />
        }
        label="Trim whitespace"
      />
    </FormControl>
  );
}
