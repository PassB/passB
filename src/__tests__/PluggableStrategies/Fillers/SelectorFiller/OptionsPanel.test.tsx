import {mount, shallow, ShallowWrapper} from 'enzyme';
import * as React from 'react';
import {OptionsPanel} from 'PluggableStrategies/Fillers/SelectorFiller/OptionsPanel';
import {Options} from 'PluggableStrategies/Fillers/SelectorFiller/SelectorFiller';
import {createTypedMap, TypedMap} from 'State/Types/TypedMap';

const getOptions = (partial: Partial<Options> = {}) => createTypedMap({
  defaultUsernameSelector: 'input[autocomplete="username"],input[type="email"]',
  defaultPasswordSelector: 'input[type="password"]',
  fillOnlyFirstMatchingField: true,
  usernameSelectorPrefix: '',
  passwordSelectorPrefix: '',
  ...partial,
});

describe('SelectorFiller OptionsPanel', () => {
  describe('rendering', () => {
    function testRendering(testOptions: TypedMap<Options>): void {
      let panel: ShallowWrapper<{}>;
      it('renders', () => {
        panel = shallow(
          <OptionsPanel options={testOptions} updateOptions={() => 0}/>,
        ).dive();
        expect(panel).toMatchSnapshot();
      });

      it('defaultUsernameSelector has correct value', () => {
        expect(panel.find('#defaultUsernameSelector').prop('value')).toBe(testOptions.get('defaultUsernameSelector'));
      });

      it('defaultUsernameSelector is not marked as erroneous', () => {
        expect(panel.find('#defaultUsernameSelector').prop('error')).toBe(false);
      });

      it('defaultPasswordSelector has correct value', () => {
        expect(panel.find('#defaultPasswordSelector').prop('value')).toBe(testOptions.get('defaultPasswordSelector'));
      });

      it('defaultPasswordSelector is not marked as erroneous', () => {
        expect(panel.find('#defaultPasswordSelector').prop('error')).toBe(false);
      });

      it('fillOnlyFirstMatchingField is checked correctly', () => {
        expect(
          (panel.find('#fillOnlyFirstMatchingField').prop('control') as JSX.Element).props.checked,
        ).toBe(testOptions.get('fillOnlyFirstMatchingField'));
      });

      it('usernameSelectorPrefix has correct value', () => {
        expect(panel.find('#usernameSelectorPrefix').prop('value')).toBe(testOptions.get('usernameSelectorPrefix'));
      });

      it('passwordSelectorPrefix has correct value', () => {
        expect(panel.find('#passwordSelectorPrefix').prop('value')).toBe(testOptions.get('passwordSelectorPrefix'));
      });
    }

    describe('default values', () => testRendering(getOptions()));
    describe('completely different options', () => testRendering(getOptions({
      defaultUsernameSelector: '',
      defaultPasswordSelector: '',
      fillOnlyFirstMatchingField: false,
      usernameSelectorPrefix: '#user',
      passwordSelectorPrefix: '#pass',
    })));
  });

  describe('should update options on interaction', () => {
    const optionsCallback = jest.fn();

    const testOptions: TypedMap<Options> = getOptions({});

    const panel = mount(
      <OptionsPanel options={testOptions} updateOptions={optionsCallback}/>,
    );

    beforeEach(() => {
      optionsCallback.mockReset();
    });

    it('updates defaultUsernameSelector', () => {
      panel.find('#defaultUsernameSelector').hostNodes().simulate('change', {target: {value: 'test'}});
      expect(optionsCallback).toHaveBeenCalledWith(testOptions.set('defaultUsernameSelector', 'test'));
    });

    it('marks defaultUsernameSelector as erroneous and does not call optionsCallback when invalid selector is entered', () => {
      panel.find('#defaultUsernameSelector').hostNodes().simulate('change', {target: {value: 'input[type="test]'}});
      expect(panel.find('#defaultUsernameSelector').first().prop('error')).toBe(true);
      expect(optionsCallback).not.toHaveBeenCalled();
    });

    it('updates defaultPasswordSelector', () => {
      panel.find('#defaultPasswordSelector').hostNodes().simulate('change', {target: {value: 'test'}});
      expect(optionsCallback).toHaveBeenCalledWith(testOptions.set('defaultPasswordSelector', 'test'));
    });

    it('marks defaultPasswordSelector as erroneous and does not call optionsCallback when invalid selector is entered', () => {
      panel.find('#defaultPasswordSelector').hostNodes().simulate('change', {target: {value: 'input[type="test]'}});
      expect(panel.find('#defaultPasswordSelector').first().prop('error')).toBe(true);
      expect(optionsCallback).not.toHaveBeenCalled();
    });

    it('updates fillOnlyFirstMatchingField', () => {
      panel.find('#fillOnlyFirstMatchingField').find('input').hostNodes()
        .simulate('change', {target: {checked: !testOptions.get('fillOnlyFirstMatchingField')}});
      expect(optionsCallback).toHaveBeenCalledWith(
        testOptions.set('fillOnlyFirstMatchingField', !testOptions.get('fillOnlyFirstMatchingField')),
      );
    });

    it('updates usernameSelectorPrefix', () => {
      panel.find('#usernameSelectorPrefix').hostNodes().simulate('change', {target: {value: 'test'}});
      expect(optionsCallback).toHaveBeenCalledWith(testOptions.set('usernameSelectorPrefix', 'test'));
    });

    it('updates passwordSelectorPrefix', () => {
      panel.find('#passwordSelectorPrefix').hostNodes().simulate('change', {target: {value: 'test'}});
      expect(optionsCallback).toHaveBeenCalledWith(testOptions.set('passwordSelectorPrefix', 'test'));
    });
  });

});
