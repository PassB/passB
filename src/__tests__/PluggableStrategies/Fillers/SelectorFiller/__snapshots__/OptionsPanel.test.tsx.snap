// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectorFiller OptionsPanel rendering completely different options renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OptionsPanelComponent
    classes={
      Object {
        "topPadding": "OptionsPanelComponent-topPadding-1",
      }
    }
    options={
      Immutable.Map {
        "defaultUsernameSelector": "",
        "defaultPasswordSelector": "",
        "fillOnlyFirstMatchingField": false,
        "usernameSelectorPrefix": "#user",
        "passwordSelectorPrefix": "#pass",
      }
    }
    updateOptions={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TextField
          error={false}
          helperText="Leave empty to only fill if there's a selector specified in the pass file"
          id="defaultUsernameSelector"
          label="Default username input css selector"
          margin="normal"
          onChange={[Function]}
          required={false}
          select={false}
          value=""
        />,
        <TextField
          error={false}
          helperText="Leave empty to only fill if there's a selector specified in the pass file"
          id="defaultPasswordSelector"
          label="Default password input css selector"
          margin="normal"
          onChange={[Function]}
          required={false}
          select={false}
          value=""
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />
          }
          id="fillOnlyFirstMatchingField"
          label="Fill only the first matching input field"
        />,
        <withStyles(Typography)
          className="OptionsPanelComponent-topPadding-1"
          gutterBottom={true}
          type="title"
        >
          Selectors from pass file
        </withStyles(Typography)>,
        <withStyles(Typography)
          type="subheading"
        >
          If the pass file contains a line starting with one of these prefixes, that selector will be used instead.
        </withStyles(Typography)>,
        <TextField
          id="usernameSelectorPrefix"
          label="username selector prefix"
          margin="normal"
          onChange={[Function]}
          required={false}
          select={false}
          value="#user"
        />,
        <TextField
          id="passwordSelectorPrefix"
          label="password selector prefix"
          margin="normal"
          onChange={[Function]}
          required={false}
          select={false}
          value="#pass"
        />,
      ],
      "fullWidth": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "error": false,
          "helperText": "Leave empty to only fill if there's a selector specified in the pass file",
          "id": "defaultUsernameSelector",
          "label": "Default username input css selector",
          "margin": "normal",
          "onChange": [Function],
          "required": false,
          "select": false,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "error": false,
          "helperText": "Leave empty to only fill if there's a selector specified in the pass file",
          "id": "defaultPasswordSelector",
          "label": "Default password input css selector",
          "margin": "normal",
          "onChange": [Function],
          "required": false,
          "select": false,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={false}
            onChange={[Function]}
          />,
          "id": "fillOnlyFirstMatchingField",
          "label": "Fill only the first matching input field",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "Selectors from pass file",
          "className": "OptionsPanelComponent-topPadding-1",
          "gutterBottom": true,
          "type": "title",
        },
        "ref": null,
        "rendered": "Selectors from pass file",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "If the pass file contains a line starting with one of these prefixes, that selector will be used instead.",
          "type": "subheading",
        },
        "ref": null,
        "rendered": "If the pass file contains a line starting with one of these prefixes, that selector will be used instead.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "id": "usernameSelectorPrefix",
          "label": "username selector prefix",
          "margin": "normal",
          "onChange": [Function],
          "required": false,
          "select": false,
          "value": "#user",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "id": "passwordSelectorPrefix",
          "label": "password selector prefix",
          "margin": "normal",
          "onChange": [Function],
          "required": false,
          "select": false,
          "value": "#pass",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TextField
            error={false}
            helperText="Leave empty to only fill if there's a selector specified in the pass file"
            id="defaultUsernameSelector"
            label="Default username input css selector"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            value=""
          />,
          <TextField
            error={false}
            helperText="Leave empty to only fill if there's a selector specified in the pass file"
            id="defaultPasswordSelector"
            label="Default password input css selector"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            value=""
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={false}
                onChange={[Function]}
              />
            }
            id="fillOnlyFirstMatchingField"
            label="Fill only the first matching input field"
          />,
          <withStyles(Typography)
            className="OptionsPanelComponent-topPadding-1"
            gutterBottom={true}
            type="title"
          >
            Selectors from pass file
          </withStyles(Typography)>,
          <withStyles(Typography)
            type="subheading"
          >
            If the pass file contains a line starting with one of these prefixes, that selector will be used instead.
          </withStyles(Typography)>,
          <TextField
            id="usernameSelectorPrefix"
            label="username selector prefix"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            value="#user"
          />,
          <TextField
            id="passwordSelectorPrefix"
            label="password selector prefix"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            value="#pass"
          />,
        ],
        "fullWidth": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "error": false,
            "helperText": "Leave empty to only fill if there's a selector specified in the pass file",
            "id": "defaultUsernameSelector",
            "label": "Default username input css selector",
            "margin": "normal",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "error": false,
            "helperText": "Leave empty to only fill if there's a selector specified in the pass file",
            "id": "defaultPasswordSelector",
            "label": "Default password input css selector",
            "margin": "normal",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={false}
              onChange={[Function]}
            />,
            "id": "fillOnlyFirstMatchingField",
            "label": "Fill only the first matching input field",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Selectors from pass file",
            "className": "OptionsPanelComponent-topPadding-1",
            "gutterBottom": true,
            "type": "title",
          },
          "ref": null,
          "rendered": "Selectors from pass file",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "If the pass file contains a line starting with one of these prefixes, that selector will be used instead.",
            "type": "subheading",
          },
          "ref": null,
          "rendered": "If the pass file contains a line starting with one of these prefixes, that selector will be used instead.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "id": "usernameSelectorPrefix",
            "label": "username selector prefix",
            "margin": "normal",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "#user",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "id": "passwordSelectorPrefix",
            "label": "password selector prefix",
            "margin": "normal",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "#pass",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SelectorFiller OptionsPanel rendering default values renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OptionsPanelComponent
    classes={
      Object {
        "topPadding": "OptionsPanelComponent-topPadding-1",
      }
    }
    options={
      Immutable.Map {
        "defaultUsernameSelector": "input[autocomplete=\\"username\\"],input[type=\\"email\\"]",
        "defaultPasswordSelector": "input[type=\\"password\\"]",
        "fillOnlyFirstMatchingField": true,
        "usernameSelectorPrefix": "",
        "passwordSelectorPrefix": "",
      }
    }
    updateOptions={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TextField
          error={false}
          helperText="Leave empty to only fill if there's a selector specified in the pass file"
          id="defaultUsernameSelector"
          label="Default username input css selector"
          margin="normal"
          onChange={[Function]}
          required={false}
          select={false}
          value="input[autocomplete=\\"username\\"],input[type=\\"email\\"]"
        />,
        <TextField
          error={false}
          helperText="Leave empty to only fill if there's a selector specified in the pass file"
          id="defaultPasswordSelector"
          label="Default password input css selector"
          margin="normal"
          onChange={[Function]}
          required={false}
          select={false}
          value="input[type=\\"password\\"]"
        />,
        <withStyles(FormControlLabel)
          control={
            <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />
          }
          id="fillOnlyFirstMatchingField"
          label="Fill only the first matching input field"
        />,
        <withStyles(Typography)
          className="OptionsPanelComponent-topPadding-1"
          gutterBottom={true}
          type="title"
        >
          Selectors from pass file
        </withStyles(Typography)>,
        <withStyles(Typography)
          type="subheading"
        >
          If the pass file contains a line starting with one of these prefixes, that selector will be used instead.
        </withStyles(Typography)>,
        <TextField
          id="usernameSelectorPrefix"
          label="username selector prefix"
          margin="normal"
          onChange={[Function]}
          required={false}
          select={false}
          value=""
        />,
        <TextField
          id="passwordSelectorPrefix"
          label="password selector prefix"
          margin="normal"
          onChange={[Function]}
          required={false}
          select={false}
          value=""
        />,
      ],
      "fullWidth": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "error": false,
          "helperText": "Leave empty to only fill if there's a selector specified in the pass file",
          "id": "defaultUsernameSelector",
          "label": "Default username input css selector",
          "margin": "normal",
          "onChange": [Function],
          "required": false,
          "select": false,
          "value": "input[autocomplete=\\"username\\"],input[type=\\"email\\"]",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "error": false,
          "helperText": "Leave empty to only fill if there's a selector specified in the pass file",
          "id": "defaultPasswordSelector",
          "label": "Default password input css selector",
          "margin": "normal",
          "onChange": [Function],
          "required": false,
          "select": false,
          "value": "input[type=\\"password\\"]",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "control": <withStyles(Checkbox)
            checked={true}
            onChange={[Function]}
          />,
          "id": "fillOnlyFirstMatchingField",
          "label": "Fill only the first matching input field",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "Selectors from pass file",
          "className": "OptionsPanelComponent-topPadding-1",
          "gutterBottom": true,
          "type": "title",
        },
        "ref": null,
        "rendered": "Selectors from pass file",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "If the pass file contains a line starting with one of these prefixes, that selector will be used instead.",
          "type": "subheading",
        },
        "ref": null,
        "rendered": "If the pass file contains a line starting with one of these prefixes, that selector will be used instead.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "id": "usernameSelectorPrefix",
          "label": "username selector prefix",
          "margin": "normal",
          "onChange": [Function],
          "required": false,
          "select": false,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "id": "passwordSelectorPrefix",
          "label": "password selector prefix",
          "margin": "normal",
          "onChange": [Function],
          "required": false,
          "select": false,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TextField
            error={false}
            helperText="Leave empty to only fill if there's a selector specified in the pass file"
            id="defaultUsernameSelector"
            label="Default username input css selector"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            value="input[autocomplete=\\"username\\"],input[type=\\"email\\"]"
          />,
          <TextField
            error={false}
            helperText="Leave empty to only fill if there's a selector specified in the pass file"
            id="defaultPasswordSelector"
            label="Default password input css selector"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            value="input[type=\\"password\\"]"
          />,
          <withStyles(FormControlLabel)
            control={
              <withStyles(Checkbox)
                checked={true}
                onChange={[Function]}
              />
            }
            id="fillOnlyFirstMatchingField"
            label="Fill only the first matching input field"
          />,
          <withStyles(Typography)
            className="OptionsPanelComponent-topPadding-1"
            gutterBottom={true}
            type="title"
          >
            Selectors from pass file
          </withStyles(Typography)>,
          <withStyles(Typography)
            type="subheading"
          >
            If the pass file contains a line starting with one of these prefixes, that selector will be used instead.
          </withStyles(Typography)>,
          <TextField
            id="usernameSelectorPrefix"
            label="username selector prefix"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            value=""
          />,
          <TextField
            id="passwordSelectorPrefix"
            label="password selector prefix"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            value=""
          />,
        ],
        "fullWidth": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "error": false,
            "helperText": "Leave empty to only fill if there's a selector specified in the pass file",
            "id": "defaultUsernameSelector",
            "label": "Default username input css selector",
            "margin": "normal",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "input[autocomplete=\\"username\\"],input[type=\\"email\\"]",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "error": false,
            "helperText": "Leave empty to only fill if there's a selector specified in the pass file",
            "id": "defaultPasswordSelector",
            "label": "Default password input css selector",
            "margin": "normal",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "input[type=\\"password\\"]",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "control": <withStyles(Checkbox)
              checked={true}
              onChange={[Function]}
            />,
            "id": "fillOnlyFirstMatchingField",
            "label": "Fill only the first matching input field",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Selectors from pass file",
            "className": "OptionsPanelComponent-topPadding-1",
            "gutterBottom": true,
            "type": "title",
          },
          "ref": null,
          "rendered": "Selectors from pass file",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "If the pass file contains a line starting with one of these prefixes, that selector will be used instead.",
            "type": "subheading",
          },
          "ref": null,
          "rendered": "If the pass file contains a line starting with one of these prefixes, that selector will be used instead.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "id": "usernameSelectorPrefix",
            "label": "username selector prefix",
            "margin": "normal",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "id": "passwordSelectorPrefix",
            "label": "password selector prefix",
            "margin": "normal",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
